app.rb

```diff
@@ -57,7 +57,8 @@ def authorized?
         url: config.idp_url,
         headers: { 'Authorization' => "Bearer #{config.attempts_api_csp_id} #{irs_attempt_api_auth_token}" }
       )
-      body = "timestamp=#{ActiveSupport::TimeZone['UTC'].now.iso8601}"
+      @today = Time.now.to_s.slice(0, 16)
+      body = "timestamp=#{Time.parse(params[:timestamp]).utc.iso8601 rescue (ActiveSupport::TimeZone['UTC'].now).iso8601}"
```

Comment:
I'm not a super big fan of the tail-end rescue syntax, especially snuck inside of a string interplation. I'd consider making a helper method for it

```suggestion
      body = "timestamp=#{parsed_timestamp(params[:timestamp])}"
```

```ruby
def parsed_timestamp(timestamp)
  Time.parse(timestamp).utc.iso8601
rescue
  ActiveSupport::TimeZone['UTC'].now.iso8601
end
```

Reply:
I did the same earlier then moved to one-liner :D